language/tools/test-generation/src/lib.rs

```diff
@@ -150,45 +150,102 @@ fn seed(seed: Option<String>) -> [u8; 32] {
     array
 }
 
-/// Run generate_bytecode for 'iterations' iterations and test each generated module
-/// on the bytecode verifier.
-pub fn run_generation(args: Args) {
-    env_logger::init();
-    let iterations = args.num_iterations;
-    let mut verified_programs: u64 = 0;
-    let mut executed_programs: u64 = 0;
-    let mut generation_options = ModuleGeneratorOptions::default();
-    generation_options.min_table_size = 10;
-    // The more structs, and the larger the number of type parameters the more complex the
-    // functions and bytecode sequences generated. Be careful about setting these parameters too
-    // large -- this can lead to expontial increases in the size and number of struct
-    // instantiations that can be generated.
-    generation_options.max_ty_params = 3;
-    generation_options.max_functions = 5;
-    generation_options.max_structs = 5;
-    generation_options.args_for_ty_params = true;
-    generation_options.references_allowed = false;
-    // Test generation cannot currently cope with resources
-    generation_options.add_resources = false;
-    let seed = seed(args.seed);
+#[derive(Debug, Clone, PartialEq)]
+pub enum Status {
+    VerificationFailure,
+    ExecutionFailure,
+    Valid,
+}
+
+fn bytecode_module(rng: &mut StdRng, module: CompiledModuleMut) -> CompiledModuleMut {
+    let mut generated_module = BytecodeGenerator::new(rng).generate_module(module.clone());
+    // Module generation can retry under certain circumstances
+    while generated_module.is_none() {
+        generated_module = BytecodeGenerator::new(rng).generate_module(module.clone());
+    }
+    generated_module.unwrap()
+}
+
+pub fn module_frame_generation(
+    num_iters: Option<u64>,
+    seed: [u8; 32],
+    sender: Sender<CompiledModuleMut>,
+    stats: Receiver<Status>,
+) {
+    let mut verification_failures: u128 = 0;
+    let mut execution_failures: u128 = 0;
+    let mut generated: u128 = 1;
+
+    let generation_options = config::module_generation_settings();
     let mut rng = StdRng::from_seed(seed);
-    for i in 0..iterations {
+    let mut module = generate_module(&mut rng, generation_options.clone()).into_inner();
+    let iters = num_iters
```

Comment:
This code is mysterious.  What is it doing?

Reply:
I'll add a comment. Since the number of iterations is now optional, this is saying that if no iteration bounds are provided, then we will only generate `u128::MAX` number of test cases.

Reply:
u128::MAX is a huge number.  Perhaps the iteration bound should not be optional.  Or use some smaller default.